src/View/ViewBuilder.php

```diff
@@ -310,7 +310,15 @@ public function getPlugin(): ?string
      */
     public function addHelper(string $helper, array $options = [])
     {
-        $this->_helpers[$helper] = $options;
+        [$plugin, $name] = pluginSplit($helper);
+        if ($plugin) {
+            $options = [
+                'class' => $helper,
```

Comment:
Shouldn't this be `className`?

Reply:
Not according to the registry code:
```php
        foreach ($objects as $i => $objectName) {
            $config = [];
            if (!is_int($i)) {
                $config = (array)$objectName;
                $objectName = $i;
            }
            [, $name] = pluginSplit($objectName);
            if (isset($config['class'])) {
                $normal[$name] = $config + ['config' => []];
            } else {
                $normal[$name] = ['class' => $objectName, 'config' => $config];
            }
        }
```

Reply:
Weird that it uses `class` when `className` is used at most places throughout the codebase.

Reply:
True. I wonder where that happened, and why docs are also not adjusted to this yet.
I see at least one part of aliasing mentioned with `className` key.

Reply:
`load()` has `'className'` it seems, but add and other parts use `'class'`.

Reply:
I think less changes would be required if we just set `$options['className'] = $helper, instead of setting the `class` and `config` keys.

Reply:
My main goal is to fade out eventually the normalization in the registry, as all such configs should be normalized already from the start, making it easier to interact with them per name key.

But if it works to simplify this using only one config key, that would of course be also nice.

Reply:
> My main goal is to fade out eventually the normalization in the registry, as all such configs should be normalized already from the start, making it easier to interact with them per name key.

`normalizeArray()` is used in `Helper` class too since helpers can use other helpers. Same for `Component`. So trying to get rid of the normalization in the registry will probably lead to code duplication.

> But if it works to simplify this using only one className config key, that would of course be also nice.

I am pretty sure this too will achieve what you want since the name is split before hand for plugin prefixed names.

Reply:
Currently you get an explicit error when a name is reused. Without your change calling `addHelper('First.Foo')` followed by `addHelper('Second.Foo')` will silently replace the 1st config which is bad IMO and can leave users scratching their heads.

Reply:
I am not sure how to otherwise mitigate the problem where you deliberatly want to use switch out a helper in a controller.
Usually our add*() does replace on config level.

I wonder if people then should keep loadHelper() here if they want it to "blow up" or if we can have a different way of differentiating this.

Reply:
Should addHelper() have a flag "overwrite" bool? default false, if false it will throw exception if already existing?

Reply:
Well you got rid of the boolean param for `setHelpers()` earlier, so adding a boolean param to `addHelper()` doesn't make sense.

A new `setHelper()` which overwrites and avoids any error being shown for conflicts sounds good. We can clarify the behaviour in the docblock.

Reply:
Sure, that could also be an option, let me try to add this.